src/uptane/tuf.h

```diff
@@ -0,0 +1,151 @@
+/**
+ * Base data types that are used in The Update Framework (TUF), part of UPTANE.
+ */
+
+#ifndef AKTUALIZR_UPTANE_TUF_H_
+#define AKTUALIZR_UPTANE_TUF_H_
+
+#include <ostream>
+#include <set>
+
+#include "crypto.h"
+
+namespace Uptane {
+
+typedef std::string KeyId;
+/**
+ * TUF Roles
+ */
+enum Role { kRoot, kSnapshot, kTargets, kTimestamp, kInvalidRole };
+
+Role RoleFromString(const std::string &rolename);
+
+std::string StringFromRole(Role role);
+
+/**
+ * Metadata version numbers
+ */
+class Version {
+ public:
+  Version() : version_(ANY_VERSION) {}
+  explicit Version(int v) : version_(v) {}
+  std::string RoleFileName(Role role) const;
+
+ private:
+  static const int ANY_VERSION = -1;
+  int version_;
+  friend std::ostream &operator<<(std::ostream &os, const Version &t);
+};
+
+class TimeStamp {
+ public:
+  static TimeStamp Now();
+  /** An invalid TimeStamp */
+  TimeStamp() {}
+  explicit TimeStamp(std::string rfc3339);
+  bool operator<(const TimeStamp &other) const;
+  bool operator>(const TimeStamp &other) const;
+  friend std::ostream &operator<<(std::ostream &os, const TimeStamp &t);
+  bool valid() const;
+
+ private:
+  std::string time_;
+};
+
+/**
+ * The hash of a file or TUF metadata.  File hashes/checksums in TUF include the length of the object, in order to
+ * defeat infinite download attacks.
+ */
+class Hash {
+ public:
+  enum Type { sha256, sha512 };
+
+  Hash(Type type, const std::string &hash);
+
+  /**
+   * Hash content and check it
+   * @param content
+   * @return
+   */
+  bool matchWith(const std::string &content) const;
+
+  bool operator==(const Hash &other) const;
+
+ private:
+  Type type_;
+  std::string hash_;
+};
+
+class Target {
+ public:
+  Target(const Json::Value &custom, const std::string &filename, Hash hash, int64_t length)
+      : custom_(custom), filename_(filename), hash_(hash), length_(length) {}
+  bool operator==(const Target &t2) const { return (filename_ == t2.filename_ && hash_ == t2.hash_); }
+
+  std::string ecu_identifier() const { return custom_["ecuIdentifier"].asString(); }
+
+  std::string filename() const { return filename_; }
```

Comment:
I would like to stick to https://google.github.io/styleguide/cppguide.html#Function_Names - I know there are quite a few deviations from that in the codebase, but we should uphold it as the standard :)